eng/pipelines/global-build.yml

```diff
@@ -92,6 +92,21 @@ jobs:
       buildArgs: -subset mono+libs /p:RuntimeFlavor=Mono
       timeoutInMinutes: 90
 
+#
+# Build Libraries AllConfigurations. This excercises the code path where we build libraries for all
+# configurations on a non Windows operating system.
+#
+- template: /eng/pipelines/common/platform-matrix.yml
+  parameters:
+    jobTemplate: /eng/pipelines/common/global-build-job.yml
+    buildConfig: debug
+    platforms:
+    - Linux_x64
+    jobParameters:
+      nameSuffix: Libraries_AllConfigurations
+      buildArgs: -subset libs -allconfigurations
+      timeoutInMinutes: 90
```

Comment:
I agree this is desirable, but any idea what it'll do to pressure on CI?

Reply:
As this skips building runtimes, hosts and other components that are built in the other dev-innerloop legs, this leg won't stand out and take more time than the others. From an overall resource perspective, this adds one more leg to the already existing 150+ legs.

Reply:
Maybe we can remove the mono legs from dev-inner loop? I know they protect the specific runtime flavor scenario, but these were added before the mobile legs existed which build that way, so maybe those are enough to protect that scenario?

Reply:
@stephentoub just got the build numbers: 25m for the allconfigurations job (vs 50+ min for the other jobs in that pipeline).

> Maybe we can remove the mono legs from dev-inner loop? I know they protect the specific runtime flavor scenario, but these were added before the mobile legs existed which build that way, so maybe those are enough to protect that scenario?

@safern sounds great. Do you wanna file an issue for that / work on that?

Reply:
I filed: https://github.com/dotnet/runtime/issues/50110